### PR TITLE
[#IOPID-1926] Service Preferences failures queue processor

### PR DESCRIPTION
**Pull request by BurnedMarshal**
#### List of Changes
Add a new queue trigger to handle service preferences failures

#### Motivation and Context
When the change feed processor for Service Preference has an error a message is sent into the failure queue. The new trigger elaborate the message retrying all the outbound flow.

#### How Has This Been Tested?

not yet

#### Types of changes

- [ ] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: AnalyticsServicePreferencesStorageQueueInboundProcessorAdapter/index.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+import { OutboundEnricher } from "../outbound/port/outbound-enricher";
+import { servicePreferencesAvroFormatter } from "../utils/formatter/servicePreferencesAvroFormatter";
+
+export type RetrievedServicePreferenceWithMaybePdvId = t.TypeOf<
+  typeof RetrievedServicePreferenceWithMaybePdvId
+>;
+const RetrievedServicePreferenceWithMaybePdvId = t.intersection([
+  RetrievedServicePreference,
+  t.partial({ userPDVId: NonEmptyString })
+]);
```
Comment (gquadrati) [inline AnalyticsServicePreferencesStorageQueueInboundProcessorAdapter/index.ts:23]: Can we import the one defined in `AnalyticsServicePreferencesChangeFeedInboundProcessorAdapter`?

File: AnalyticsServicePreferencesStorageQueueInboundProcessorAdapter/index.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+const servicePreferencesConfig = withTopic(
+  config.servicePreferencesKafkaTopicConfig.SERVICE_PREFERENCES_TOPIC_NAME,
+  config.servicePreferencesKafkaTopicConfig
+    .SERVICE_PREFERENCES_TOPIC_CONNECTION_STRING
+)(config.targetKafka);
+
+const servicePreferencesTopic = {
+  ...config.targetKafka,
+  messageFormatter: servicePreferencesAvroFormatter()
+};
```
Comment (gquadrati) [inline AnalyticsServicePreferencesStorageQueueInboundProcessorAdapter/index.ts:36]:
```suggestion
const servicePreferencesTopic = {
  ...servicePreferencesConfig,
  messageFormatter: servicePreferencesAvroFormatter()
};
```

**Review by gquadrati — CHANGES_REQUESTED** [review]

File: AnalyticsServicePreferencesStorageQueueInboundProcessorAdapter/index.ts (hunk 1 above)
Comment (BurnedMarshal) [reply AnalyticsServicePreferencesStorageQueueInboundProcessorAdapter/index.ts]: Fixed in [19d305e](https://github.com/pagopa/io-functions-elt/pull/85/commits/19d305e58b9073b3625d858bd3ce34a608fdd700)

File: AnalyticsServicePreferencesStorageQueueInboundProcessorAdapter/index.ts (hunk 2 above)
Comment (BurnedMarshal) [reply AnalyticsServicePreferencesStorageQueueInboundProcessorAdapter/index.ts]: Fixed in [19d305e](https://github.com/pagopa/io-functions-elt/pull/85/commits/19d305e58b9073b3625d858bd3ce34a608fdd700)

**Review by gquadrati — APPROVED** [review]
LGTM

**Pull request merged**